Sources/EduIDExpansion/Flows/PersonalInfo/PersonalInfoViewController.swift

```diff
@@ -9,13 +10,54 @@ class PersonalInfoViewController: UIViewController, ScreenWithScreenType {
     // - delegate
     weak var delegate: PersonalInfoViewControllerDelegate?
     
+    var viewModel: PersonalInfoViewModel
+    
+    private var stack: UIStackView!
+    static var indexOfFirstLinkedAccount = 5
+    
+    //MARK: - init
+    init(viewModel: PersonalInfoViewModel) {
+        self.viewModel = viewModel
+        super.init(nibName: nil, bundle: nil)
+        
+        viewModel.dataAvailableClosure = { [weak self] model in
+            self?.setupUI(model: model)
+        }
+        
+        viewModel.serviceRemovedClosure = { [weak self] account in
+            let views = self?.stack.arrangedSubviews.filter { view in
+                (view as? ActionableControlWithCollapsibleBody)?.institution == account.schacHomeOrganization
+            } as? [UIView] ?? []
+            UIView.animate(withDuration: 0.2, delay: 0, animations: {
+                views.forEach { view in
+                    view.isHidden = true
+                }
+            }) { [weak self] _ in
+                views.forEach { view in
+                    self?.stack.removeArrangedSubview(view)
+                }
+            }
+        }
+        
+        viewModel.dataFetchErrorClosure = { [weak self] error in
+            let alert = UIAlertController(title: "Error", message: error.localizedDescription, preferredStyle: .alert)
+            self?.present(alert, animated: true)
+            DispatchQueue.main.asyncAfter(deadline: .now() + 3) {
```

Comment:
Auto dismissal of an error seems like unexpected behavior to me. Was this a requirement?

Reply:
good remark, I will add an ok-button